src/anomalib/data/kolektor.py

```diff
@@ -0,0 +1,179 @@
+    ), "Mismatch between anomalous images and ground truth masks. Make sure the mask files in 'ground_truth' \
+              folder follow the same naming convention as the anomalous images in the dataset (e.g. image: '000.png', \
+              mask: '000.png' or '000_mask.png')."
+
+    # Get the dataframe for the required split
+    if split:
+        samples = samples[samples.split == split].reset_index(drop=True)
+
+    return samples
+
+
+class KolektorDataset(AnomalibDataset):
+    def __init__(
+        self,
+        task: TaskType,
+        transform: A.Compose,
+        root: Path | str,
+        split: str | Split | None = None,
+    ) -> None:
+        super().__init__(task=task, transform=transform)
+
+        self.root = root
+        self.split = split
+
+    def _setup(self) -> None:
+        self.samples = make_kolektor_dataset(self.root, split=self.split)
+
+
+class Kolektor(AnomalibDataModule):
+    def __init__(
+        self,
+        root: Path | str,
+        image_size: int | tuple[int, int] | None = None,
+        center_crop: int | tuple[int, int] | None = None,
+        normalization: str
+        | InputNormalizationMethod = InputNormalizationMethod.IMAGENET,
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        num_workers: int = 8,
+        task: TaskType = TaskType.SEGMENTATION,
+        transform_config_train: str | A.Compose | None = None,
+        transform_config_eval: str | A.Compose | None = None,
+        test_split_mode: TestSplitMode = TestSplitMode.FROM_DIR,
+        test_split_ratio: float = 0.2,
+        val_split_mode: ValSplitMode = ValSplitMode.SAME_AS_TEST,
+        val_split_ratio: float = 0.5,
+        seed: int | None = None,
+    ) -> None:
+        super().__init__(
+            train_batch_size=train_batch_size,
+            eval_batch_size=eval_batch_size,
+            num_workers=num_workers,
+            test_split_mode=test_split_mode,
+            test_split_ratio=test_split_ratio,
+            val_split_mode=val_split_mode,
+            val_split_ratio=val_split_ratio,
+            seed=seed,
+        )
+
+        self.root = Path(root)
+
+        transform_train = get_transforms(
+            config=transform_config_train,
+            image_size=image_size,
+            center_crop=center_crop,
+            normalization=InputNormalizationMethod(normalization),
+        )
+        transform_eval = get_transforms(
+            config=transform_config_eval,
+            image_size=image_size,
+            center_crop=center_crop,
+            normalization=InputNormalizationMethod(normalization),
+        )
+
+        self.train_data = KolektorDataset(
+            task=task,
+            transform=transform_train,
+            split=Split.TRAIN,
+            root=root,
+        )
+        self.test_data = KolektorDataset(
+            task=task,
+            transform=transform_eval,
+            split=Split.TEST,
+            root=root,
+        )
```

Comment:
This is missing `prepare_data` method. Have a look at https://github.com/openvinotoolkit/anomalib/blob/27adbaeb7da406c52bceba06dbc875ba3c0e955a/src/anomalib/data/mvtec.py#L258 to see an example implementation